extensions-contrib/thrift-extensions/src/main/java/io/druid/data/input/thrift/ThriftInputRowParser.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.data.input.thrift;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.metamx.common.IAE;
+import com.metamx.common.parsers.Parser;
+import io.druid.data.input.InputRow;
+import io.druid.data.input.MapBasedInputRow;
+import io.druid.data.input.impl.InputRowParser;
+import io.druid.data.input.impl.ParseSpec;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TSerializer;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TSimpleJSONProtocol;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * 1. load thrift class from classpath or provided jar
+ * 2. deserialize content bytes and serialize to json
+ * 3. use JsonSpec to do things left
+ */
+public class ThriftInputRowParser implements InputRowParser<Object>
+{
+  private final ParseSpec parseSpec;
+  private final String jarPath;
+  private final String thriftClassName;
+  private final String protocol;
+
+  final private TDeserializer deserializer;
+  final private TSerializer serializer = new TSerializer(new TSimpleJSONProtocol.Factory());
+  final private Parser<String, Object> parser;
+  volatile private Class<TBase> thriftClass = null;
+
+  @JsonCreator
+  public ThriftInputRowParser(
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("thriftJar") String jarPath,
+      @JsonProperty("thriftClass") String thriftClassName,
+      @JsonProperty("protocol") String protocol
+  )
+  {
+    this.protocol = protocol;
+
+    if (protocol == null || protocol.equals("compact")) {
```

Comment:
can we use Jackson to deserialize the protocol into a TCompactProtocol or TBinaryProtocol? this iwll avoid the if/else check